tests/gateways/test_jlap.py

```diff
@@ -278,6 +320,8 @@ def test_jlap_sought(
     package_repository_base: Path,
 ):
     """Test that we try to fetch the .jlap file."""
+    (package_repository_base / "osx-64" / "repodata.jlap").unlink(missing_ok=True)
```

Comment:
Is this just a precaution? Or does this point to a larger issue of test cleanup failing elsewhere?

Reply:
The test web server thread points to a single directory. We could use other strategies like copying those files into a per-test directory or creating a per-test scoped fixture to cleanup.